database/apikeys.go

```diff
@@ -42,13 +44,32 @@ type (
 	}
 )
 
+// New creates a random new API key.
+func NewAPIKey() APIKey {
+	return APIKey(base32.HexEncoding.WithPadding(base32.NoPadding).EncodeToString(fastrand.Bytes(PubKeySize)))
+}
+
+// Bytes defines the way we decode an API key.
+func (ak APIKey) Bytes() ([]byte, error) {
+	return base32.HexEncoding.WithPadding(base32.NoPadding).DecodeString(strings.ToUpper(string(ak)))
+}
+
 // IsValid checks whether the underlying string satisfies the type's requirement
-// to represent a []byte with length PubKeySize which is encoded as base64URL.
+// to represent a []byte with length PubKeySize which is encoded as base32 with
+// no padding.
 // This method does NOT check whether the API exists in the database.
 func (ak APIKey) IsValid() bool {
-	b := make([]byte, PubKeySize)
-	n, err := base64.URLEncoding.Decode(b, []byte(ak))
-	return err == nil && n == PubKeySize
+	b, err := ak.Bytes()
+	return err == nil && len(b) == PubKeySize
+}
+
+// LoadBytes encodes a []byte of size PubKeySize into an API key.
+func (ak *APIKey) LoadBytes(b []byte) error {
+	if len(b) != PubKeySize {
+		return errors.New("key too short")
```

Comment:
it might be too long 🤷‍♂️  I would error out in the form of `unexpected key size, %v != %v` etc

Reply:
Major *facepalm*...

Yes, yes it can... I don't know what I was thinking. I'll use your suggestion.